spec/integration/friendship_management_spec.rb

```diff
@@ -0,0 +1,39 @@
+require 'rails_helper'
+
+RSpec.describe 'Friendship management', type: :feature do
+  let(:user) { User.create!(name: 'Mohammad', email: 'jay@gmail.com', password: 'abc123456789') }
+  let(:friend) { User.create!(name: 'James', email: 'fahad@gmail.com', password: 'abc123456789') }
```

Comment:
You are testing different things here. 

Please create one test for checking the login process, another for the sign in, and another for the posts.
Also, add some integration tests to check the process of sending a friendship invite, accepting it, and denying it.
Remember each test has to go in a different file.